modules/oci-publish/02_mod.mk

```diff
@@ -0,0 +1,124 @@
+# Copyright 2023 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Utility functions
+fatal_if_undefined = $(if $(findstring undefined,$(origin $1)),$(error $1 is not set))
+oci_digest = $(shell head -1 $(oci_digest_path_$1) 2> /dev/null)
+sanitize_target = $(subst :,-,$1)
+registry_for = $(firstword $(subst /, ,$1))
+
+# Utility variables
+current_makefile = $(lastword $(MAKEFILE_LIST))
+current_makefile_directory = $(dir $(current_makefile))
+
+# Validate globals that are required
+$(call fatal_if_undefined,bin_dir)
+$(call fatal_if_undefined,build_names)
+
+# Set default config values
+RELEASE_DRYRUN ?= false
+CRANE_FLAGS ?= # empty by default
+COSIGN_FLAGS ?= # empty by default
+OCI_SIGN_ON_PUSH ?= true
+
+# Default variables per build_names entry
+#
+# $1 - build_name
+define default_per_build_variables
+release_dryrun_$1 ?= $(RELEASE_DRYRUN)
+crane_flags_$1 ?= $(CRANE_FLAGS)
+cosign_flags_$1 ?= $(COSIGN_FLAGS)
+oci_sign_on_push_$1 ?= $(OCI_SIGN_ON_PUSH)
+endef
+
+$(foreach build_name,$(build_names),$(eval $(call default_per_build_variables,$(build_name))))
+
+# Validate variables per build_names entry
+#
+# $1 - build_name
+define check_per_build_variables
+$(call fatal_if_undefined,oci_digest_path_$1)
+$(call fatal_if_undefined,oci_layout_path_$1)
+$(call fatal_if_undefined,oci_image_name_$1)
```

Comment:
Can we keep the old name here, that will make it easier to migrate and test these changes? `oci_$1_image_name`